executor/set.go

```diff
@@ -203,13 +211,13 @@ func (e *SetExecutor) setSysVariable(name string, v *expression.VarAssignment) e
 			valStr, err = value.ToString()
 			terror.Log(err)
 		}
-		if name != variable.AutoCommit {
-			logutil.BgLogger().Info("set session var", zap.Uint64("conn", sessionVars.ConnectionID), zap.String("name", name), zap.String("val", valStr))
-		} else {
-			// Some applications will set `autocommit` variable before query.
-			// This will print too many unnecessary log info.
-			logutil.BgLogger().Debug("set session var", zap.Uint64("conn", sessionVars.ConnectionID), zap.String("name", name), zap.String("val", valStr))
-		}
+	}
+	if name != variable.AutoCommit {
+		logutil.BgLogger().Info(fmt.Sprintf("set %s var", scopeStr), zap.Uint64("conn", sessionVars.ConnectionID), zap.String("name", name), zap.String("val", valStr))
```

Comment:
Does this change close #17091?

Reply:
Yeah, I think so.
```
mysql> set @@tidb_hash_join_concurrency=2;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> set @@global.tidb_hash_join_concurrency=2;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```


```
[2020/07/09 11:11:50.981 +08:00] [INFO] [set.go:216] ["set session var"] [conn=1] [name=tidb_hash_join_concurrency] [val=2]
[2020/07/09 11:12:00.222 +08:00] [INFO] [set.go:216] ["set global var"] [conn=1] [name=tidb_hash_join_concurrency] [val=2]

```

Reply:
@niedhui Would you like to help put this change in an individual pull request so it can be cherry-picked to older release?

Reply:
done. https://github.com/pingcap/tidb/pull/18457